lib/backends/wasm/ops/instance-normalization.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {InstanceNormalization} from '../../../ops/instance-normalization';
+import {Tensor} from '../../../tensor';
+import {WasmBinding} from '../../../wasm-binding';
+import {WasmInferenceHandler} from '../inference-handler';
+
+export class WasmInstanceNormalization extends InstanceNormalization {
+  run(inferenceHandler: WasmInferenceHandler, inputs: Tensor[]): Tensor[] {
+    const x = inputs[0];
+    const scale = inputs[1];
+    const b = inputs[2];
+
+    // calculate channel size (i.e.) data points per channel
+    let channelSize = 1;
+    for (let i = 2; i < x.dims.length; i++) {
+      channelSize *= x.dims[i];
+    }
+
+    // create output Tensor after determining output size
+    const y = new Tensor(x.dims, x.type);
+    WasmBinding.getInstance().ccall(
+        '_instance_normalization_f32', [x.floatData, 'float32ptr'], [y.floatData, 'float32ptr', 'out'],
+        [x.dims[0], 'int32'], [x.dims[1], 'int32'], [channelSize, 'int32'], [scale.floatData, 'float32ptr'],
+        [b.floatData, 'float32ptr'], [this.epsilon, 'float32']);
+
+    return [y];
+  }
+
+  // overriding the checkInputTypes() in the base class because Wasm backend has special type limitations
+  checkInputTypes(inputs: Tensor[]): boolean {
```

Comment:
can we apply the special type check only? I think typescript keyword `super` can help to call parent class method.

Reply:
Right now `checkInputTypes()` in the base class also does some other validation other than type validation and hence we need to copy over that snippet of code as well in this method as this method overrides completely the base method. We can make a change later to make it in such a way that only type check is overridden and other kinds of input validation is still common across base and derived classes (hence prevent code duplication). But there are one or two more ops that require this kind of change. So I prefer to do this separately in another change.